nntrainer/src/model_loader.cpp

```diff
@@ -240,8 +242,10 @@ int ModelLoader::loadFromIni(std::string ini_file, NeuralNetwork &model) {
   status = loadModelConfigIni(ini, model);
   NN_INI_RETURN_STATUS();
 
-  status = loadDatasetConfigIni(ini, model);
-  NN_INI_RETURN_STATUS();
+  if (model.isTrain()) {
+    status = loadDatasetConfigIni(ini, model);
```

Comment:
Given there is no data set for the model in inference mode, are you planning to manually feed in the data bypassing the dataset? or the dataset must be set externally in this case?

Reply:
dataset is not needed I guess for inference.

Reply:
as discussed offline, the data will be passed directly.